stdeb/util.py

```diff
@@ -1152,13 +1155,18 @@ def __init__(self,
         else:
             self.binary_target_lines = ''
 
+        self.changelog_distname = CHANGELOG_PY2_DISTNAME%self.__dict__
         if with_python2:
             self.control_py2_stanza = CONTROL_PY2_STANZA%self.__dict__
         else:
             self.control_py2_stanza = ''
 
         if with_python3:
             self.control_py3_stanza = CONTROL_PY3_STANZA%self.__dict__
+            if self.distname3:
+                if with_python2:
+                    raise ValueError("Suites are shared between versions. To use Suite3 run --with-python3 true --with-python2 false only.")
```

Comment:
This only needs to raise if `self.distname3` is different from `self.distname`, no? Otherwise it should be fine to continue with either of them (to not change backward compatibility I would suggest with the Python 2 name).

Reply:
With this PR as is, if someone adds a Suite3 option and then runs with python2 and python3 stdeb will raise an error right away.

With your suggestion, if someone adds a Suite3 option matching their Suite value, they'll get no error until they change one and not the other.

If it were me, I'd want the error to occur as soon as I add the conflicting options, not potentially hidden until the options are actually used to create different behavior between python2 and python3. But I will yield to preference here you prefer the error held back as long as possible.

Reply:
Being able to explicitly list the `Suite3` option even if it isn't different sounds like a good case to support. Why should we not allow that but require the user to remove the `Suite3` option if the value ever aligns with `Suite`?

Reply:
> Why should we not allow that but require the user to remove the `Suite3` option if the value ever aligns with `Suite`

It would only be disallowed when python2 and python3 are being run together. There is no problem if both config values are identical but you're building them in separate invocations.

It's just a personal preference, but if I'm "allowed" to enter two config values, but only if they are both identical, I would prefer to be forced to just use one value unless I actually need them to be different.

The way I see it, there just isn't a wide use case supported by allowing both to be specified but only if they're identical.

* Projects who want to use the same suites for both may continue using the Suite option and building with-python2 and with-python3 together

* Projects who want to build for distinct suites will have to build with-python2 and with-python3 separately most of the time, so it's only when the two values happen to be equivalent that they would be able to make use of the relaxed error condition.

I'll go ahead and make the change though because I don't think there's anything *wrong* with your proposal, it's just not what I would expect.

Reply:
f673824 relaxes the error.